### PR TITLE
More error information is needed wrt. remote files

### LINKED ISSUE #1735: More error information is needed wrt. remote files - [Errno 2] No such file
**Setup**
`dvc --version` = 0.32.1+db5a0 and local and remote server uses ubuntu 18.04.2. For method of installation see https://github.com/PeterFogh/dvc_dask_use_case.

**Problem**
When remote files (via SSH in my case) are missing when running `dvc repro`, because `dvc checkout` have not been performed, yet. DVC puts out the following error message: `Error: failed to reproduce '[DVC_STAGE_FILE]': [Errno 2] No such file`. I find this error message easy to misunderstand. I wish the error message pointed to the specific remote file that is missing. In my case I know it is `Posts.tsv`, because I have removed it my self.

There is the verbose log from executing `dvc repro &> no_error.log` of https://github.com/PeterFogh/dvc_dask_use_case reprository, with no error:
[no_error.log](https://github.com/iterative/dvc/files/2974913/no_error.log)

And there is the verbose log with the error from executing `dvc repro &> remove_post_tsv.log` after I have removed the remote file with `rm Posts.tsv`:
[remove_posts_tsv.log](https://github.com/iterative/dvc/files/2974918/remove_posts_tsv.log)

I will trying to come up with a PR which contains the logging fo the file path.

### PR DESCRIPTION
**Pull request by PeterFogh**
As described in #1735, I think DVC needs to output more info about remote files,
Thus, I have added this to `stage.py`.
Additionally, I think that DVC should not crash if the SSH client is trying to remove a fil which already does not exists. This, I have implemented the catch of the `FileNotFoundError : [Errno 2] No such file` exception.

Fixes #1735

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -27,3 +27,6 @@ innosetup/config.ini
 dvc/version.py
 
 *.swp
+
+pip-wheel-metadata/
+.vscode/
```
Comment (shcheklein) [inline .gitignore]: minor: let's keep a newline please

File: dvc/command/repro.py [hunk 2]
```diff
@@ -80,7 +80,7 @@ def add_parser(subparsers, parent_parser):
         "-c",
         "--cwd",
         default=os.path.curdir,
-        help="Directory within your repo to reroduce from.",
+        help="Directory within your repo to reproduce from.",
```
Comment (shcheklein) [inline dvc/command/repro.py]: 👍 thanks!

File: dvc/remote/ssh/connection.py [hunk 3]
```diff
@@ -143,7 +144,16 @@ def walk_files(self, directory):
 
     def remove(self, path):
         self._sftp_connect()
-        self._sftp.remove(path)
+        try:
+            self._sftp.remove(path)
+        except FileNotFoundError as exc:
```
Comment (shcheklein) [inline dvc/remote/ssh/connection.py]: @mroutis it's not Python 2 compatible, right? is true for sftp client as well that it throws IoError on python 2 instead?

File: dvc/stage.py [hunk 4]
```diff
@@ -253,10 +255,13 @@ def changed(self):
...
-        """
-        Used mainly for `dvc remove --outs`
-        """
+        """Used mainly for `dvc remove --outs` and :func:`Stage.reproduce`."""
         for out in self.outs:
+            logger.debug(
+                "Removing output '{out}' of '{stage}'.".format(
+                    out=out, stage=self.relpath
+                )
+            )
```
Comment (shcheklein) [inline dvc/stage.py]: ❤️ thanks!

File: dvc/remote/ssh/connection.py [hunk 5]
```diff
@@ -143,7 +144,16 @@ def walk_files(self, directory):
...
+            if exc.errno == errno.ENOENT:
+                logger.warning(
```
Comment (shcheklein) [inline dvc/remote/ssh/connection.py]: Also, we should be using the same operation semantics for all remotes, I think. Can we confirm that all of them do not throw if path does not exist? Alternative to this is to actually ensure this semantic on the caller side. Wrap FileNotFoundError (IoError) here into DvcNotFoundException and catch upstream. Your thoughts, @efiop @mroutis ?

Comment (efiop) [reply dvc/remote/ssh/connection.py]: @shcheklein This approach is fine, this is how we handle it in local remote too. Other remotes don't have the same behaviour though, we'll need to fix that in the future.

File: dvc/stage.py [hunk 6]
```diff
@@ -210,8 +210,11 @@ def _changed_deps(self):
         for dep in self.deps:
             if dep.changed():
```
Comment (efiop) [inline dvc/stage.py]: We could simply use dep.status() here and extract the message from it, without calling changed() first.

File: dvc/stage.py [hunk 7]
```diff
@@ -222,8 +225,11 @@ def _changed_outs(self):
         for out in self.outs:
             if out.changed():
```
Comment (efiop) [inline dvc/stage.py]: Same here.

Comment (PeterFogh) [reply dvc/stage.py]: @efiop, I have changed the code to only use status() ones. Are there anything else that needs to be changed? :)

**Comment by efiop** [comment]
@PeterFogh Thanks! :slightly_smiling_face:  Could you also address @shcheklein 's comments above please? (about missing newline  at the end of the file and FileNotFoundError not supported by python2).

File: dvc/remote/ssh/connection.py (hunk 3 above)
Comment (efiop) [reply dvc/remote/ssh/connection.py]: Yes, we should use IOError and check for exc.errno. That would work for both py2 and py3.

File: dvc/remote/ssh/connection.py (hunk 5 above)
Comment (efiop) [reply dvc/remote/ssh/connection.py]: Same here https://github.com/iterative/dvc/pull/1737#discussion_r266248094 .

File: .gitignore (hunk 1 above)
Comment (PeterFogh) [reply .gitignore]: Of course :)

**Comment by PeterFogh** [comment]
@efiop, I assume that I do not need to install a local Python 2 env. for test version 2? I hope Travis will test it?

**Comment by efiop** [comment]
Thank you @PeterFogh ! :tada:

**Comment by PeterFogh** [comment]
Thanks for making and showing me the last needed changes.

**Comment by efiop** [comment]
@PeterFogh Sorry, fixed a few small things, since I just couldn't wait to try out these amazing changes and merge them :slightly_smiling_face:

**Pull request merged**